### PR TITLE
Connect Post Office to VNET

### PR DESCRIPTION
**Pull request by kristianschneider**
This PR prepares for the integration into a vnet.
It creates the appropriate private endpoints and the required a-record DNS entries for a cosmosDB and other resources.

This PR covers https://app.zenhub.com/workspaces/mighty-ducks---the-outlaws-6193fe815d79fc0011e741b1/issues/energinet-datahub/the-outlaws/111

As it says on the tin: Do not merge into main before the VNET is ready

### CODE REVIEW

File: build/infrastructure/main/func-subdomain.tf [hunk 1]
```diff
@@ -30,7 +33,7 @@ module "func_subdomain" {
     WEBSITES_ENABLE_APP_SERVICE_STORAGE       = true
     FUNCTIONS_WORKER_RUNTIME                  = "dotnet-isolated"
     # Endregion
-    MESSAGES_DB_CONNECTION_STRING             = local.message_db_connection_string
+    MESSAGES_DB_CONNECTION_STRING             = local.MESSAGES_DB_CONNECTION_STRING
```
Comment (Renetnielsen) [inline build/infrastructure/main/func-subdomain.tf:36]: Nice fix 💪

File: build/infrastructure/main/variables.tf [hunk 2]
```diff
@@ -44,4 +44,19 @@ variable shared_resources_keyvault_name {
 variable shared_resources_resource_group_name {
   type          = string
   description   = "Name of the Resource Group, that contains the shared resources."
+}
+
+variable "subnet_internal_private_endpoints_address_space" {
+  type          = string
+  description   = "Address space of the subnet for internal private endpoints"
```
Comment (Renetnielsen) [inline build/infrastructure/main/variables.tf:51]: `~~private endpoints~~ -> Private endpoints`

This is a resource, so it should be capital cased like a name

File: build/infrastructure/main/variables.tf [hunk 3]
```diff
@@ -44,4 +44,19 @@ variable shared_resources_keyvault_name {
...
+}
+
+variable "subnet_external_private_endpoints_address_space" {
+  type          = string
+  description   = "Address space of the subnet for external private endpoints"
```
Comment (Renetnielsen) [inline build/infrastructure/main/variables.tf:56]: `~~private endpoints~~ -> Private endpoints`

This is a resource, so it should be capital cased like a name

File: build/infrastructure/main/variables.tf [hunk 4]
```diff
@@ -44,4 +44,19 @@ variable shared_resources_keyvault_name {
...
+}
+
+variable "subnet_vnet_integrations_functions_address_space" {
+  type          = string
+  description   = "Address space of the subnet for Virtual Network Integrations of Azure Functions"
```
Comment (Renetnielsen) [inline build/infrastructure/main/variables.tf:61]: `~~Virtual Network Integrations~~ -> Virtual Network integrations`

Same way as azure names it, with lower cased i

**Review by Renetnielsen — DISMISSED** [review]
LGTM small description change

**Review by Renetnielsen — DISMISSED** [review]

**Review by Renetnielsen — DISMISSED** [review]

**Review by djorgensendk — APPROVED** [review]

**Review by Renetnielsen — APPROVED** [review]

**Pull request merged**